components/event-service/nats/server.go

```diff
@@ -105,8 +108,11 @@ func spawnNatsInternalServer(c *config.EventConfig, m *multiEmbeddedServer) erro
 
 	m.runInGoroutine(func() error {
 		m.registerServer(ns)
-		ns.Start()
-		return errors.New("NATS server exited")
+		go ns.Start()
```

Comment:
since we are adding the statements to stop the control, do we still need to define it as a routine?
can you test by running without routine.